test/parsing/issue-21710.good

```diff
@@ -0,0 +1,2 @@
+issue-21710.chpl:1: In module 'my_freader':
+issue-21710.chpl:5: error: fields cannot specify generic array types
```

Comment:
I don't have a problem with this going in as-is (esp. since it's such a big improvement over the current behavior).  If I were fine-tuning it, I would try to tailor the error message to be more specific about the problem at hand—i.e., I could imagine a new user not understanding what it was about `x: [] int` that made it generic (and have had savvy Arkouda developers ask this kind of question in settings like `export proc foo(x: [] int)` before, for example...).  So messages like the following seem like they'd be much more helpful, if sufficiently easy to produce:

* `x: [] int` or `x: [?D] int`: "array fields must declared in terms of a domain (index set)"
* `x: [1..n] ?t`: "array fields must have a well-defined element type"  (if I'm correct that this case isn't currently supported)

It'd be good to have a test that checked for the non-field / standalone variable case as well.

Reply:
What I'll do for for this now (as a sort of IOU) is convert this into a named error e.g,. `CHPL_REPORT(PartiallySpecifiedArrayType)`, and stuff all the information in it we could possibly need to print all these different cases. Then in a future PR I can can focus on fine-tuning all the error message wordings.

Reply:
It'd also be fine with me to just add an issue to fine-tune this as an IOU rather than code changes if you prefer.